### ISSUE TITLE
MTP k>1 crashes during warmup due to uninitialized embed_tokens weights

### ISSUE THREAD

**Issue by QwertyJack** [jd-opensource/xllm#835]
## Description

When using MTP (Multi-Token Prediction) with `num_speculative_tokens > 1` (e.g., k=4), the server crashes during warmup.

## Environment

- Model: GLM-4.7 + GLM-4.7-MTP
- Device: NPU (Ascend)
- Config: `--num_speculative_tokens 4`

## Root Cause

1. **Shared word_embedding causes ATB state corruption**: When k>1, the draft model's `forward()` is called k times per step, each calling the shared `NpuWordEmbeddingImpl`. The layer's internal state gets corrupted during rapid repeated calls.

2. **MTP model doesn't load embed_tokens weights**: `mtp_model_base.h::load_state_dict()` doesn't load `embed_tokens_` weights, even though `mtp_layer_parameters.safetensors` contains `model.embed_tokens.weight`. When we stop sharing the word_embedding layer, the MTP model uses uninitialized weights.

## Crash Log

```
I20260130 01:38:20.409365 profile_manager.cpp:809] Warming up prefill request: tokens=5000
[ERROR] TBE Subprocess[task_distribute] raise error[], main process disappeared!
```

## Stack Trace

```
#0 xllm::layer::NpuWordEmbeddingImpl::build_node_variant_pack()
#1 xllm::layer::NpuWordEmbeddingImpl::forward()
#2 xllm::MtpModelImplBase<xllm::Glm4MoeDecoderLayer>::forward()
```

## Fix

1. Don't share word_embedding layer when `num_speculative_tokens > 1`
2. Add `embed_tokens_` to `load_state_dict()`, `verify_loaded_weights()`, `merge_loaded_weights()` in `mtp_model_base.h`
